### PR TITLE
feat: Add authentication

### PR DESCRIPTION
**Pull request by dantehemerson**
Still in Progress

### CODE REVIEW

File: src/modules/users/application/services/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { authConfig } from '@config/auth.config';
+import { IAuthService } from '@modules/users/domain/interfaces/auth-service.interface';
+import { IJWTPayload } from '@modules/users/domain/interfaces/jwt-payload.interface';
+import jwt from 'jsonwebtoken';
+
+export class AuthService implements IAuthService {
+	public signJWT(payload: IJWTPayload): string {
+		return jwt.sign(payload, authConfig.jwtSecret);
+	}
+}
```
Comment (dantehemerson) [inline src/modules/users/application/services/auth.service.ts:10]: I have a doubt here, Do you think this should be a provider instead?

**Review by stevescruz — APPROVED** [review]

File: src/config/auth.config.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const authConfig = {
+	jwtSecret: process.env.JWT_SECRET ?? 'secret',
+};
```
[the remark this answers is not in the document]
Comment (stevescruz) [reply src/config/auth.config.ts:3]: Yes, agree

File: src/modules/users/application/services/auth.service.ts (hunk 1 above)
Comment (stevescruz) [reply src/modules/users/application/services/auth.service.ts:10]: The signJWT method could come from a provider, so we use it through an interface

Comment (dantehemerson) [reply src/modules/users/application/services/auth.service.ts]: Done, moved to a provider

**Pull request merged**